File: Core/Components/Framework/ComponentGroupBL.cs
using Core.Components.Forms;
using Core.Extensions;
using Core.Models;

namespace Core.Components.Framework
{
    public class ComponentGroupBL : PopupEditor
    {
        private Component ComGroupEntity => Entity as Component;
        public ComponentGroupBL() : base(nameof(Component))
        {
            Name = "ComponentGroup";
            Title = "Section properties";
            Icon = "fa fa-wrench";
            PopulateDirty = false;
            DOMContentLoaded += AlterPosition;
        }

        private void AlterPosition()
        {
            Element.ParentElement.AddClass("properties");
        }
    }
}

File: CoreAPI/wwwRoot/js/core/forms/componentGroupBL.js
import { EditForm } from "../editForm.js";
import { Component } from "../models/component.js";

/**
* ComponentBL class
* @class
* @extends EditForm
*/
export class ComponentGroupBL extends EditForm {
    /**
    * Constructor
    */
    constructor() {
        super('Component');
        this.Name = "ComponentEditor";
        this.Title = "Component properties";
        this.Icon = "fa fa-wrench";
        this.Id = "EditComponent_" + this.Id;
        this.Entity = new Component();
        this.PopulateDirty = false;
        this.DOMContentLoaded += this.AlterPosition.bind(this);
    }

    /**
    * Alter position
    */
    AlterPosition() {
        this.Element.parentElement.ToggleClass("properties");
    }
}
